src/test/scala/mesosphere/marathon/api/akkahttp/v2/GroupsControllerTest.scala

```diff
@@ -239,4 +245,26 @@ class GroupsControllerTest extends UnitTest with ScalatestRouteTest with Inside
 
     val groupsController: GroupsController = new GroupsController(electionService, infoService, groupManager, groupApiService, config)
   }
+
+  case class FixtureWithRealGroupManager(
```

Comment:
Why is the mock not working for us? Should we just go with the actual group manager in get rid off the mocked one?

Reply:
hmm maybe... ? groupsresource actually had also two kinds of tests - with real and mocked one. 

The reason why I need a real one here is that part of the behavior is injected to the groupManager via function as parameter and I need to test that behavior so I need to actually execute the method...

Reply:
btw. I would probably stick with this two fixture setup - the problem is, that some methods  calling group repository on the background, and those are just more easily mocked than done on real group manager... But this concrete thing (with the injected function) I cannot easily test on mock. It is a mess (kind of)... :(

Reply:
I'd opt for not mocking at all then 🙂.

Reply:
A [Hangout](https://www.youtube.com/watch?v=z9quxZsLcfo&list=PLJb2p0qX8R_qSRhs14CiwKuDuzERXSU8m) with DHH, Kent Beck and Martin Fowler convinced me that we should not mock too much. Why not call the real thing?

Reply:
(disclaimer - I did not watch the video yet, will do that later today)

Reply:
Hm, the name is confusing then. What is bad about mocking the repository instead of the manager?

Reply:
let's continue this discussion on slack - I am still interested in your input even though I am merging this...